server/src/main/java/org/elasticsearch/action/search/SearchTransportAPMMetrics.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.search;
+
+import org.elasticsearch.telemetry.TelemetryProvider;
+import org.elasticsearch.telemetry.metric.LongHistogram;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+public class SearchTransportAPMMetrics {
```

Comment:
This comment explains the purpose of this class: https://github.com/elastic/elasticsearch/pull/101577/files#r1379132909

Reply:
I can see why we ended up here. It currently feels a bit weird to have one class that includes this single instance member. Why isn't the instance member directly added to the only consumer of this new class? perhaps we are planning on adding more later? Not a big deal for me, just trying to understand the choice.

Reply:
It _is_ weird I agree.  The root of the issue is that if you double register a metric, it will throw an exception.  This is not an APM thing, but a choice made/enforced by the folks to implemented our library on APM. Otherwise it would have been safe to just register a metric in a constructor then store a reference to it in a member variable, as you describe.  

So instead, we need to ensure that the lifecycle of the metric is tied to the lifecycle of the node / MeterRegistry.  One way of doing that it to hold the metric inside a wrapper class that guarantees the metric is already registered, then pass that wrapper into other spots where the lifecycle is not tied to the node's (or is but could change in future).  It's a lot of boilerplate and I don't love it, but it's better than exceptions that only present themselves under class-reloading/second-instantiation at runtime.

WDYT?

Reply:
Put another way, this purpose of this class is not for organization but instead to help enforce correctness.